src/handlers/tokenSearch.ts

```diff
@@ -56,4 +57,21 @@ const tokenSearch = async (searchParams: {
   }
 };
 
-export default tokenSearch;
+export const walletFilter = async (params: {
+  addresses: EthereumAddress[];
+  fromChainId: number;
+  toChainId: number;
+}) => {
+  try {
+    const { addresses, fromChainId, toChainId } = params;
+    const filteredAddresses = await tokenSearchApi.post(`/${fromChainId}`, {
+      addresses,
+      toChainId,
+    });
+    return filteredAddresses?.data?.data || [];
+  } catch (e) {
+    logger.error(
```

Comment:
Should we be returning a empty or null value here

Reply:
Thanks for highlighting this. It caused me to realize that we were not handling the failure case well at all (if this call blows up for some reason all tokens would be unavailable for swap).

I decided to let the error bubble up and handle this failure case at the call site, by allowing all tokens to be swapped rather than none. This seems like the lesser of two evils.

Reply:
Yeah sounds good to me 👍